corva/network/api.py

```diff
@@ -1,134 +1,111 @@
-import os
+import posixpath
 import re
-from typing import List, Optional
+from typing import Optional
 
-from requests import Response, Session
-from requests.adapters import HTTPAdapter
-from urllib3 import Retry
+import requests
+import requests.adapters
+import urllib3
 
 
-class Api:
+class Api(requests.Session):
     """Provides a convenient way to access Corva API and Corva Data API
 
-    Api is a thin wrapper around `requests` library that handles
-     authorization, adds timeouts and retries to request.
+    Api is a thin wrapper around `requests.Session` that adds
+     authorization, timeouts and retries to requests.
     """
 
-    ALLOWED_METHODS = {'GET', 'POST', 'PATCH', 'PUT', 'DELETE'}
-    DEFAULT_TIMEOUT = 600
-    DEFAULT_MAX_RETRIES = 3
+    MAX_RETRIES = 3
+    TIMEOUT = 30  # seconds
 
     def __init__(
-         self,
-         api_url: str,
-         data_api_url: str,
-         api_key: str,
-         app_name: str,
-         timeout: Optional[int] = None,
-         max_retries: Optional[int] = None
+        self,
+        api_url: str,
+        data_api_url: str,
+        api_key: str,
+        app_name: str,
+        timeout: Optional[int] = None,
+        max_retries: Optional[int] = None,
     ):
-        self.timeout = timeout or self.DEFAULT_TIMEOUT
-        self.max_retries = max_retries or self.DEFAULT_MAX_RETRIES
+        super().__init__()
+
+        self.timeout = timeout or self.TIMEOUT
+        self.max_retries = max_retries or self.MAX_RETRIES
         self.api_url = api_url
         self.data_api_url = data_api_url
         self.api_key = api_key
         self.app_name = app_name
-        self.session = self._init_session(
-            api_key=api_key,
-            app_name=app_name,
-            max_retries=self.max_retries,
-            allowed_methods=list(self.ALLOWED_METHODS)
-        )
 
-    @staticmethod
-    def _init_session(api_key: str, app_name: str, max_retries: int, allowed_methods: List[str]):
-        session = Session()
+        self.headers.update(
+            {
+                'Authorization': f'API {api_key}',
+                'X-Corva-App': app_name,
+            }
+        )
 
-        session.headers.update({
-            'Authorization': f'API {api_key}',
-            'X-Corva-App': app_name
-        })
-        session.mount(
+        self.mount(
             'https://',
-            HTTPAdapter(
-                max_retries=Retry(
-                    total=max_retries,
+            requests.adapters.HTTPAdapter(
```

Comment:
maybe leave it to users to implement retries?

Reply:
yep